core/monitoring/system_monitor.py

```
@@ -0,0 +1,512 @@
+                "python": sys.version.split()[0],
+                "cuda": self._get_cuda_version(),
+                "platform": sys.platform
+            }
+        }
+        return info
+    
+    def _get_cpu_model(self) -> str:
+        """Get CPU model name."""
+        try:
+            if sys.platform == "linux":
+                result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call with `shell=True` seems safe, but may be changed in the future; consider rewriting without `shell`

(S602)